pytest_rerunfailures.py

```diff
@@ -173,6 +182,21 @@ def _remove_failed_setup_state_from_session(item):
     setup_state.stack = list()
 
 
+def _should_hard_fail_on_error(session_config, report):
+    if report.outcome != 'failed':
```

Comment:
It seems to me, that this test is also done in line 230 with `report.failed` or am I missing some edge case here? Which test case would fail, if we leave it out?

Reply:
This is unrelated since it says that anything that is not failed cannot be a hard failure. Which makes sense to me, since otherwise this function would potentially flag non failed things as hard failures

Reply:
I understand the reason but I do not see, why we need it twice currently. It seems to be a guard for a case, that is not possible at the moment. But I would like to refactor line 230 anyhow, as this is not so easy to comprehend any more.

Reply:
I’m working on [this issue](https://github.com/pytest-dev/pytest-rerunfailures/issues/43) and in the change I have a refactor of that line.

This change came out of it. It’s just unrelated, so I raised it first.